content/blog/hacking-build-process.md

```diff
@@ -0,0 +1,344 @@
+---
+date: 2025-04-28
+title: 'Hacking The Build and Deployment Process of Serverless Workflows'
+---
+
+# Hacking The Build and Deployment Process of Serverless Workflows
+
+In this guide, we'll dive under the hood of the serverless workflow build process, examine how it works internally, and learn how to take control of it when needed.
+
+## The Problem
+
+We're working with a [repository](https://github.com/masayag/poc-kafka-logic-operator) that contains an issue with the standard workflow build script described in our [previous post](./building-and-deploying-workflows.md). When running the script, you'll encounter [an error](https://github.com/apache/incubator-kie-tools/issues/3084) - the manifests can't be generated due to an incompatible workflow spec, specifically with the `eventRef` element.
+
+Interestingly, if you run the workflow using `mvn clean quarkus:dev`, it starts successfully. This reveals a discrepancy between:
+
+- The sonataflow-runtimes library that runs the workflow
+- The spec definition used by the kn-workflow v1.35 tool to generate manifests
+
+Instead of using the standard script, we'll explore a workaround to deploy the [lock-flow workflow](https://github.com/masayag/poc-kafka-logic-operator/blob/main/callback-flow/src/main/resources/lock.sw.yaml) to an OCP cluster using the gitops profile.
+
```

Comment:
It may be nice to have a `Prerequisites` section to list what are the expectations:
* have kn CLI installed (so you can remove the sentence from the next section)
* have a kafka cluster running on an OCP cluster
* have RHDH and the orchestrator installed

Some may be dumb/obvious but at least everything is clearly put on the table